### PR TITLE
Address-book

### LINKED ISSUE #829: Build a Contact Book With Python, PyQt, and SQLite
Build-a-Contact-Book_Watermarked

### Demo: A Contact Book With Python
# Project Overview
> Prerequisites
- Step 1: Creating the Contact Book’s Skeleton App With PyQt
   Structuring the Contact Book Project
   Creating the Application’s Main Window
   Coding and Running the Application
- Step 2: Building the Contact Book’s GUI With Python
- Step 3: Setting Up the Contact Book’s Database
   Connecting to the Database With PyQt and SQLite
   Creating the contacts Table
   Testing the Contact Book’s Database
- Step 4: Displaying and Updating Existing Contacts
   Creating a Model to Handle the Contact Data
   Connecting the Model to the View
   Displaying and Updating Contacts
- Step 5: Creating New Contacts
  Creating the Add Contact Dialog
  Launching the Add Contact Dialog
  Processing the Add Dialog’s Data in the Model
- Step 6: Deleting Existing Contacts
  Deleting Selected Contacts
  Clearing the Contact Database
> Conclusion

### PR DESCRIPTION
**Pull request by sameeksharl**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [ ] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [x] Others

Title: Address-book

Folder: Address-book

Requirements: None

Script: addressbook.py

Arguments: none

Contributor: sameeksharl

Description: This project is to create an Address book in which the user can add a new contact, edit and delete existing contact and view all the contacts.

### CODE REVIEW

**Comment by sameeksharl** [comment]
@kaustubhgupta Could you please review the pr?

File: Address-Book/Readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+
```
Comment (kaustubhgupta) [inline Address-Book/Readme.md:2]: remove these blank lines

File: Address-Book/Readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# ADDRESS-BOOK
+
+[![forthebadge](https://forthebadge.com/images/badges/made-with-python.svg)](https://forthebadge.com)
+
+This project is to create an Address book using python in which the user can add a new contact, edit and delete existing contact and view all the contact using Tkinter and SQlite.
+Tkinter library for rendering graphics and SQlite is used for database.
+
+
+# Setup instructions
```
Comment (kaustubhgupta) [inline Address-Book/Readme.md:11]: All subheadings after Address-Book heading should be level 2 (##)

File: Address-Book/Readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+* To install the library tkinter we use the pip install command in the command prompt:
+	pip install tkinter
```
Comment (kaustubhgupta) [inline Address-Book/Readme.md:13]: tkinter comes preinstalled with python, you can remove this section

File: Address-Book/Readme.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+# Usage
+* In this python project, the user has to click on a button which functions the user wants to access
```
Comment (kaustubhgupta) [inline Address-Book/Readme.md:16]: Can you rewrite it? 
```suggestion
* In this python project, the user has button functionality to trigger that specific function.
```

File: Address-Book/addressbook.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        list_of_names.append(student[1])
+        for j in range(len(student)):
+            e = Entry(root, width=11, fg='Gray20') 
+            e.grid(row=i, column=j) 
+            e.insert(END, student[j])
+        i=i+1
+'''
+Getting the path of database and defining the table to be created
+'''
+database = r"addressbook.db"
```
Comment (kaustubhgupta) [inline Address-Book/addressbook.py:141]:
```suggestion
database = r"./Address-Book/addressbook.db"
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
- Flash a message when a contact is added or deleted (not in terminal)
- When contact is deleted, it still shows in the view all list. 
- Other changes:

**Comment by sameeksharl** [comment]
@kaustubhgupta its deleted only if we click on 'view all' again after deleting right??
I didn't get this issue ? could u pls tell me again so that i can rectify ??

**Comment by sameeksharl** [comment]
Could u pls check now??? @kaustubhgupta

**Review by kaustubhgupta — APPROVED** [review]

**Comment by sameeksharl** [comment]
Thank you for your response!!

**Review by santushtisharma10 — APPROVED** [review]
LGTM

**Comment by sameeksharl** [comment]
@kaustubhgupta  @avinashkranjan could u pls merge the pr??
as @avinashkranjan also said LGTM

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta @avinashkranjan could u pls merge the pr??
> as @avinashkranjan also said LGTM

PA (avinash) is yet to review the PR. He will merge the PR soon or on the last date of GSSOC. Kindly have patience. You will get your points for PR.

**Comment by sameeksharl** [comment]
Thanks for the info !!! not about points, just a verification .Wanted make sure it is reviewed coz some other project’s mentors are not verifying either its been a month . Sent from Mail for Windows 10 From: Kaustubh GuptaSent: 26 May 2021 14:55To: avinashkranjan/Amazing-Python-ScriptsCc: sameeksharl; AuthorSubject: Re: [avinashkranjan/Amazing-Python-Scripts] Address-book ***@***.*** @avinashkranjan could u pls merge the pr??as @avinashkranjan also said LGTMPA (avinash) is yet to review the PR. He will merge the PR soon or on the last date of GSSOC. Kindly have patience. You will get your points for PR.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Review by avinashkranjan — APPROVED** [review]
LGTM..

**Pull request merged**